Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrash.m

```diff
@@ -306,41 +316,64 @@ - (void)deleteAllReports {
     [self.crashReportStore deleteAllFiles];
 }
 
+- (NSArray<BugsnagThread *> *)captureThreads:(NSException *)exc depth:(int)depth {
+    NSArray *addresses = [exc callStackReturnAddresses];
+    int numFrames = (int) [addresses count];
+    uintptr_t *callstack = malloc(numFrames * sizeof(*callstack));
+
+    for (NSUInteger i = 0; i < numFrames; i++) {
+        callstack[i] = [addresses[i] unsignedLongValue];
+    }
+    if (numFrames > 0) {
+        depth = 0; // reset depth if the stack does not need to be generated
```

Comment:
This took some parsing in my head!

I think it would make the intent clearer if the work to populate callstack from the exception happens inside the `numFrames > 0` path, then we don't need to realloc and some comments can describe the two cases separately.

Reply:
Refactored this so that callstack allocation happens within the `numFrames > 0` path.